Introduce program linkage (#295)
microsoft/hlsl-specs: specs/language/basic.tex

\begin{grammar}
  \define{translation-unit}\br
  \opt{declaration-sequence}
\end{grammar}

\p A \textit{program} is one or more translation units \textit{linked} together.
A program built from a single translation unit, bypassing a linking step is
called \textit{freestanding}.

\p A program is said to be \textit{fully linked}, when it contains no
\textit{unresolved external} declarations, and all \textit{exported}
declarations are entry point declarations (\ref{Basic.start}). A program is said
to be \textit{partially linked}, when it contains at least one unresolved
external declaration or at least one exported declaration that is not an entry
point.

\p An implementation may generate programs as fully linked or partially linked
as requested by the user, and a runtime may allow fully linked or partially
linked programs as the implementation allows.

\p A name has \textit{linkage} if it can refer to the same entity as a name
introduced by a declaration in another scope. If a variable, function, or
another entity with the same name is declared in several scopes, but does not
have sufficient \textit{linkage}, then several instances of the entity are
generated.

\begin{itemize}

> \item If a name has \textit{no linkage}, it may not be referred to by names from

Reviewer: ```suggestion
\item A name with \textit{no linkage}, may not be referred to by names from
```

Consider modifying each of the items to either be of the form "If a name has ..." or "A name with ...", for consistency in wording.
